### ISSUE TITLE
Azure datalake support in dask

### ISSUE THREAD

**Issue by martindurant** [dask/dask#2777]
[azure.datalake.store](https://github.com/Azure/azure-data-lake-store-python) provides pythonic access to bytes stored on Azure.

The following file provides a shim to enable dask to access azure in parallel, given urls like `'adl://bucket/path/to/key'`: https://github.com/oxygenanywhere/dask/blob/master/dask/bytes/adl.py , putting it on par with hdfs, s3, gcs.

This module is anecdotally believed to be working. However, we currently have no way to test this, since we have no Azure credentials. Furthermore, it's not clear where the module should live. If within dask itself, then some testing machinery (such as VCR) would need to be pulled in. Perhaps the module would find a more obvious place in the ADL repo.

Currently, the filesystem instance will pass credential strings between processes, which is potentially insecure - although perhaps this is less severe for applications presumed to be running within Azure. It may be better to think of the best way to securely provide a source of credentials to each worker.

cc @teoliphant 

xref #2776

**Comment by mrocklin** [comment]
@martindurant can I ask you to provide an entry in the docs pointing to the adl support, perhaps with a suitable "experimental" note?  That way we can close this out, and hopefully get more eyes on the project.

**Comment by martindurant** [comment]
Yes, will do. For reference, https://github.com/dask/dask-adlfs is what you need.

**Comment by martindurant** [comment]
(that shim could live in fsspec in the future...)
